third_party/move/move-compiler/src/inlining/translate.rs

```diff
@@ -345,7 +349,14 @@ impl<'l, 'r> SubstitutionVisitor<'l, 'r> {
                         items.push_back(sp(loc, SequenceItem_::Seq(body)));
                         Some(UnannotatedExp_::Block(items))
                     },
-                    _ => panic!("ICE expected function parameter to be a lambda"),
+                    _ => {
+                        self.inliner.env.add_diag(diag!(
```

Comment:
Why that? This was an ICE for a reason before (as it supposed to not happen). Do you have evidence this need to be an error like this?

Reply:
We should never produce an ICE, but this ICE happens if I pass a function parameter from one inlined method to another.  There's no logical reason why that shouldn't work, but implementing it is nontrivial.  That case is unsupported functionality.